### ISSUE TITLE
EmptyClassBlock should not be reported for objects deriving from superclasses

### ISSUE THREAD

**Issue by silmeth** [arturbosch/detekt#707]
When creating an anonymous object being instance of an abstract class that has no abstract methods, one needs to explicitly write out the empty block of the implementation, which causes ` EmptyClassBlock` to be reported by detekt.

Such an empty block is valid and pretty useful eg. when creating generic type tokens, like the ones for use with Jackson:

```kotlin
val listOfMapsToken = object : TypeReference<List<Map<String, Any?>>>() { /* a necessary empty block */ }
val deserialized: List<Map<String, Any?>> = objectMapper.readValue("[{\"foo\": \"bar\"}, {\"foo2\": null}]", listOfMapsToken)
```

This usage can be seen in [jackson-module-kotlin](https://github.com/FasterXML/jackson-module-kotlin/blob/master/src/main/kotlin/com/fasterxml/jackson/module/kotlin/Extensions.kt#L18).
Thus this is a false positive – this block actually serves a purpose, it lets Kotlin compiler correctly parse the code. ;-)

**Comment by arturbosch** [comment]
Yep, good catch. This reporting is like a dejavu :).

**Comment by arturbosch** [comment]
Which detekt version do you use?
This is already fixed in rc6-2.

**Comment by silmeth** [comment]
My bad, I didn’t check it, as it was switched on recently and I believed we had current-at-the time master build. Seems that’s not true and we’re currently on ancient 0.3. Will close the issue then, and upgrade detekt on our Sonar.

Thanks for the answer and sorry for the confusion. :)

**Comment by arturbosch** [comment]
Oh yes, sonar-kotlin can't be upgraded for rc6 for now.
The sonar gradle packaging plugin breaks packaging with newer kotlin version and migration to maven sucks :(
